src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -179,5 +206,52 @@ public class Utils {
         r.addAction(new ExecutionModelAction(stages))
     }
 
+    /**
+     * Creates and sets the loading for a cache of {@link Describable}s descending from the given descriptor type.
+     *
+     * @param type The {@link Descriptor} class whose extensions we want to find.
+     * @param includeClassNames Optionally include class names as keys. Defaults to false.
+     * @param excludedSymbols Optional list of symbol names to exclude from the cache.
+     * @return A {@link LoadingCache} for looking up types from symbols.
+     */
+    static generateTypeCache(Class<? extends Descriptor> type, boolean includeClassNames = false,
```

Comment:
Yeah, my attitude is that the type cache gets refreshed every 10 minutes and that's good enough for me.